server/src/test/java/org/opensearch/common/cache/tier/TieredSpilloverCacheTests.java

```diff
@@ -0,0 +1,837 @@
+            onHeapCacheSize,
+            diskCacheSize,
+            eventListener,
+            0
+        );
+
+        int numOfItems1 = randomIntBetween(onHeapCacheSize + 1, totalSize);
+        List<String> onHeapKeys = new ArrayList<>();
+        List<String> diskTierKeys = new ArrayList<>();
+        for (int iter = 0; iter < numOfItems1; iter++) {
+            String key = UUID.randomUUID().toString();
+            if (iter > (onHeapCacheSize - 1)) {
+                // All these are bound to go to disk based cache.
+                diskTierKeys.add(key);
+            } else {
+                onHeapKeys.add(key);
+            }
+            LoadAwareCacheLoader<String, String> loadAwareCacheLoader = getLoadAwareCacheLoader();
+            tieredSpilloverCache.computeIfAbsent(key, loadAwareCacheLoader);
+        }
+
+        for (int iter = 0; iter < numOfItems1; iter++) {
+            if (randomBoolean()) {
+                if (randomBoolean()) {
+                    int index = randomIntBetween(0, onHeapKeys.size() - 1);
+                    assertNotNull(tieredSpilloverCache.get(onHeapKeys.get(index)));
+                } else {
+                    int index = randomIntBetween(0, diskTierKeys.size() - 1);
+                    assertNotNull(tieredSpilloverCache.get(diskTierKeys.get(index)));
+                }
+            } else {
+                assertNull(tieredSpilloverCache.get(UUID.randomUUID().toString()));
+            }
+        }
+        assertEquals(numOfItems1, tieredSpilloverCache.count());
+    }
+
+    public void testWithDiskTierNull() throws Exception {
+        int onHeapCacheSize = randomIntBetween(10, 30);
+        MockCacheEventListener<String, String> eventListener = new MockCacheEventListener<String, String>();
+
+        StoreAwareCacheBuilder<String, String> onHeapCacheBuilder = new MockOnHeapCache.Builder<String, String>().setMaxSize(
+            onHeapCacheSize
+        );
+        TieredSpilloverCache<String, String> tieredSpilloverCache = new TieredSpilloverCache.Builder<String, String>()
+            .setOnHeapCacheBuilder(onHeapCacheBuilder)
+            .setListener(eventListener)
+            .build();
+
+        int numOfItems = randomIntBetween(onHeapCacheSize + 1, onHeapCacheSize * 3);
+        for (int iter = 0; iter < numOfItems; iter++) {
+            LoadAwareCacheLoader<String, String> loadAwareCacheLoader = getLoadAwareCacheLoader();
+            tieredSpilloverCache.computeIfAbsent(UUID.randomUUID().toString(), loadAwareCacheLoader);
+        }
+        assertTrue(eventListener.enumMap.get(CacheStoreType.ON_HEAP).evictionsMetric.count() > 0);
+        assertEquals(0, eventListener.enumMap.get(CacheStoreType.DISK).cachedCount.count());
+        assertEquals(0, eventListener.enumMap.get(CacheStoreType.DISK).evictionsMetric.count());
+        assertEquals(0, eventListener.enumMap.get(CacheStoreType.DISK).missCount.count());
+    }
+
+    public void testPut() {
```

Comment:
how about testing below:
1. Fill the onheap cache with entries.
2. Add onHeap cache capacity amount of new items in tiered cache.
3. Validate all items added in step 2 is only part of the onheap cache.

Reply:
Added as part of new test.